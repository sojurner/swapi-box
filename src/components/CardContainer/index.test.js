import React from "react";

import { CardContainer } from "./index";

describe("CardContainer", () => {
  let wrapper;
  let mockData;
  let toggleFavorites;
  let savedFavorites;
  let handlePage;

  beforeEach(() => {
    mockData = [{name: "Luke Skywalker", species: "Human", homeworld: "Tatooine", population: "200000", favorite: false}]
    toggleFavorites = jest.fn()
    savedFavorites = []
    handlePage = jest.fn()

    wrapper = shallow(
    <CardContainer 
      data={mockData}
      handlePage={handlePage} 
      toggleFavorites={toggleFavorites}
      savedFavorites={savedFavorites}
    />);

  })

  it("should match snapshot with data object passed", () => {
    
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot without data object passed", () => {
    mockData = []
    expect(wrapper).toMatchSnapshot();
  });
});