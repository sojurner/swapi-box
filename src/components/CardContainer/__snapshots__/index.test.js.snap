// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot with data object passed 1`] = `
<section
  className="card_container"
>
  <Component
    card={
      Object {
        "favorite": false,
        "homeworld": "Tatooine",
        "name": "Luke Skywalker",
        "population": "200000",
        "species": "Human",
      }
    }
    key="0"
    toggleFavorites={[Function]}
  />
  <button
    className="previous-page-button"
    onClick={[Function]}
  />
  <button
    className="next-page-button"
    onClick={[Function]}
  />
</section>
`;

exports[`CardContainer should match snapshot with favorites 1`] = `
<section
  className="card_container"
>
  <Component
    card={
      Object {
        "favorite": true,
        "homeworld": "Tatooine",
        "name": "Luke Skywalker",
        "population": "200000",
        "species": "Human",
      }
    }
    key="0"
    toggleFavorites={[Function]}
  />
  <button
    className="previous-page-button"
    onClick={[Function]}
  />
  <button
    className="next-page-button"
    onClick={[Function]}
  />
</section>
`;

exports[`CardContainer should match snapshot without data object passed 1`] = `
<section
  className="card_container"
/>
`;
